utils/status.go

```diff
@@ -0,0 +1,244 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"strconv"
+	"time"
+
+	"github.com/application-stacks/runtime-component-operator/common"
+	routev1 "github.com/openshift/api/route/v1"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	servingv1 "knative.dev/serving/pkg/apis/serving/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type replicasStatus struct {
+	ReadyReplicas   int32
+	UpdatedReplicas int32
+}
+
+func (r *ReconcilerBase) CheckComponentStatus(ba common.BaseComponent) (reconcile.Result, error) {
+	// Create status condition if Component Ready condition type does not exit
+	s := ba.GetStatus()
+	sc := s.GetCondition(common.StatusConditionTypeComponentReady)
+	if sc == nil {
+		sc = s.NewCondition()
+		sc.SetType(common.StatusConditionTypeComponentReady)
+		sc.SetStatus(corev1.ConditionUnknown)
+		sc.SetMessage("")
+	}
+
+	status, message, reason := corev1.ConditionFalse, "", ""
+
+	scReconciled := s.GetCondition(common.StatusConditionTypeReconciled)
+	scReady := s.GetCondition(common.StatusConditionTypeResourceReady)
+
+	// Check reconciled and resource status
+	if scReconciled != nil && scReconciled.GetStatus() == corev1.ConditionTrue && scReady != nil && scReady.GetStatus() == corev1.ConditionTrue {
+		status = corev1.ConditionTrue
+		message = "RuntimeComponent is reconciled and resources are ready"
```

Comment:
Suggestion: Application is reconciled and resources are ready

Using Application will avoid reference to RuntimeComponent in OLO/WLO